contracts/Bridge.sol

```diff
@@ -21,48 +21,64 @@ contract Bridge {
         gateway = _gateway;
     }
 
-    function lock() external payable {
-		require(msg.value > 0, "amount is too low");
+    function lock(address _tokenAddress) external payable checkAmount(msg.value) {
         nonce++;
-		emit LogETHLocked(msg.sender, msg.value, nonce);
+		emit LogETHLocked(_tokenAddress, msg.sender, msg.value, nonce);
 	}
 
-    function claim(address tokenAddress, uint amount, address to, uint _nonce) external onlyGateway {
-        require(to != address(0), "cannot trasfer to zero address");
-        require(amount > 0, "amount is too low");
-        require(processedNonces[_nonce] == false, "transaction already processed");
+    function claim(address _tokenAddress, uint _value, address _recepient, uint _nonce) external onlyGateway checkAddress(_recepient) checkStatus(_nonce) checkAmount(_value) {
         processedNonces[_nonce] = true;
 
-        if (!tokenExistance[tokenAddress]) {
-            tokenList[tokenAddress] = IERC20Mintable(tokenAddress);
-            tokenExistance[tokenAddress] = true;
+        address wrappedTokenAddress = nativeToWrapped[_tokenAddress];
+        WrappedToken wrappedTokenInstance;
+
+        if (wrappedTokenAddress != address(0)) {
+            wrappedTokenInstance = WrappedToken(wrappedTokenAddress);
+        } else {
+            wrappedTokenInstance = new WrappedToken();
+            nativeToWrapped[_tokenAddress] = address(
+                wrappedTokenInstance
+            );
         }
 
-        tokenList[tokenAddress].mint(to, amount);
+        wrappedTokenInstance.mint(_recepient, _value);
 
-        emit LogTokenMinted(tokenAddress, to, amount, _nonce);
+        emit LogTokenMinted(nativeToWrapped[_tokenAddress], _recepient, _value, _nonce);
     }
     
-    function burnFrom(address tokenAddress, address  owner, address spender, uint256 value, uint256 deadline, uint8 v, bytes32 r, bytes32 s) external onlyGateway {
-        require(value > 0, "amount is too low");
-		tokenList[tokenAddress].burnFrom(owner, spender, value, deadline, v, r, s);
+    function burnFrom(address _tokenAddress, address  _owner, uint256 _value) external onlyGateway checkAddress(_owner) checkAmount(_value)  {
```

Comment:
You don't need an `_owner` parameter: `msg.sender` is the address that you need.